### PR TITLE
DOC: added examples in spherical Bessel docstrings

### PR DESCRIPTION
**Pull request by lorenzophys**
#### Reference issue

https://github.com/scipy/scipy/issues/7168

#### What does this implement/fix?

I added examples and reference to spherical Bessel docstrings in scipy.special.

#### Additional information

Locally the documentation renders correctly.

### CODE REVIEW

**Review by rgommers — COMMENTED** [review]
Thanks @lorenzophys, this is looking pretty good. Just a comment about the relations for derivatives.

File: scipy/special/_spherical_bessel.py [hunk 1]
```diff
@@ -155,7 +267,8 @@ def spherical_in(n, z, derivative=False):
 
 
 def spherical_kn(n, z, derivative=False):
-    r"""Modified spherical Bessel function of the second kind or its derivative.
+    r"""Modified spherical Bessel function of the second kind or its
```
Comment (person142) [inline scipy/special/_spherical_bessel.py:270]: Better to leave this as a single line per PEP 257:

https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

Comment (lorenzophys) [reply scipy/special/_spherical_bessel.py:270]: Sure thing!

**Review by person142 — APPROVED** [review]
Looks good; restarted CI to see if that fixes the failure. (Ping me if it's green o/w I'm likely to forget.)

**Comment by person142** [comment]
Thanks @lorenzophys!

**Pull request merged**